create-nex-server/src/create-nex-server.ts

```diff
@@ -0,0 +1,246 @@
+		shortName = fullName;
+	}
+
+	if (!program.nexVersion) {
+		let typedValue = ''; // * Allow the user to type in a custom version not in the list
+		const response = await prompts({
+			type: 'autocomplete',
+			name: 'version',
+			message: 'NEX version:',
+			choices: NEX_VERSIONS.map(version => ({ title: version, value: version })),
+			onRender() {
+				typedValue = (this as any).input; // TODO - Augment the prompts type defs
+			}
+		}, promptOptions);
+
+		if (typeof response.version === 'string') {
+			program.nexVersion = response.version.trim();
+		} else {
+			typedValue = typedValue.trim();
+
+			if (semver.valid(typedValue)) {
+				program.nexVersion = typedValue;
+			}
+		}
+	}
+
+	if (!program.nexVersion) {
+		console.log(
+			'\nPlease specify a valid (semver) NEX version:\n' +
+			`    ${command} ${green('--nex-version=<version>')}\n` +
+			'For example:\n' +
+			`    ${command} ${green('--nex-version=3.10.0')}\n\n` +
+			`Run ${command} ${green('--help')} to see all options.`
+		);
+
+		process.exit(1);
+	}
+
+	if (!program.accessKey) {
+		const response = await prompts({
+			type: 'text',
+			name: 'key',
+			message: 'Server access key (e.g, ridfebb9):'
+		}, promptOptions);
+
+		if (typeof response.key === 'string') {
+			program.accessKey = response.key.trim();
+		}
+	}
+
+	if (!program.accessKey) {
+		console.log(
+			'\nPlease specify a server access key:\n' +
+			`    ${command} ${green('--access-key=<key>')}\n` +
+			'For example:\n' +
+			`    ${command} ${green('--access-key=ridfebb9')}\n\n` +
+			`Run ${command} ${green('--help')} to see all options.`
+		);
+
+		process.exit(1);
+	}
+
+	if (!program.sessionKeySize) {
+		let typedValue = ''; // * Allow the user to type in a custom size not in the list
+		const response = await prompts({
+			type: 'autocomplete',
+			name: 'size',
+			message: 'Session key size:',
+			choices: [
+				{ title: '32 bytes', value: 32 },
+				{ title: '16 bytes', value: 16 }
+			],
+			onRender() {
+				typedValue = (this as any).input; // TODO - Augment the prompts type defs
+			}
+		}, promptOptions);
+
+		if (typeof response.size === 'number') {
+			program.sessionKeySize = response.size;
+		} else {
+			const size = parseInt(typedValue.trim());
+
+			if (size) {
+				program.sessionKeySize = size;
+			}
+		}
+	}
+
+	if (!program.sessionKeySize) {
+		console.log(
+			'\nPlease specify a session key size:\n' +
+			`    ${command} ${green('--session-key-size=<size>')}\n` +
+			'For example:\n' +
+			`    ${command} ${green('--session-key-size=32')}\n\n` +
+			`Run ${command} ${green('--help')} to see all options.`
+		);
+
+		process.exit(1);
+	}
+
+	if (!program.fragmentSize) {
+		let typedValue = ''; // * Allow the user to type in a custom size not in the list
+		const response = await prompts({
+			type: 'autocomplete',
+			name: 'size',
+			message: 'Fragment size:',
+			choices: [
+				{ title: '1300 bytes', value: 1300 },
+				{ title: '900 bytes', value: 900 }
```

Comment:
We could maybe add some hint about which value *should* be used depending on the PRUDP or NEX version. Like this:
`1300 bytes (default for PRUDP v1)`
`900 bytes (default for PRUDP v0)`

Reply:
That's a good point :+1: